File: cpp/198_HouseRobber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int prev_max = 0, curr_max = 0;
        
        for (int n : nums) {
            int temp = curr_max;
            curr_max = max(prev_max + n, curr_max);
            prev_max = temp;
        }
        return curr_max;
    }
};

File: kotlin/198_HouseRobber.kt
import kotlin.math.max
class Solution {
    fun rob(nums: IntArray): Int {
        var prev: Int = 0
        var curr: Int = 0
        
        for (n in nums) {
            val temp: Int = curr
            curr = max(prev + n, curr)
            prev = temp
        }
        return curr
    }
}
